op_builder/cpu/cpu_adam.py

```diff
@@ -12,6 +12,13 @@ class CPUAdamBuilder(CPUOpBuilder):
 
     def __init__(self):
         super().__init__(name=self.NAME)
+        self.dtype = None
+
+    def set_dtype(self, dtype):
+        import torch
+        assert (dtype in [torch.bfloat16, torch.half, torch.float32])
```

Comment:
I think this check would be better in `cxx_args()` and other compilation code paths, even if repeated.

Reply:
removed set_dtype. no need to pass dtype during build time.